### PR TITLE
WIP: Move big parts of cmd_log_send into libkb

### PR DESCRIPTION
**Pull request by MarcoPolo**
@patrickxb 

This is a wip, but let me know if I'm on the right track.

Basically I want to move log send out of client so we can call it from mobile.

### CODE REVIEW

File: go/libkb/log_send.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+		return err
+	}
+	if err := gz.Close(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Contextified) post(status, kbfsLog, svcLog, desktopLog, updaterLog, startLog string) (string, error) {
```
Comment (patrickxb) [inline go/libkb/log_send.go]: I'd make a new type and not add anything to libkb.Contextified.  That's a special type...

**Comment by gabriel** [comment]
IMO, we should avoid Contextified, and GlobalContext at all costs (should only be used for very top level stuff)... If this is only using Log and API, can we avoid Contextified?

For example,
```go
func post(api libkb.API, ......., log logger.Logger) (string, error) {
}
```

It reduces the dependency situation *immensely*

**Comment by gabriel** [comment]
ok nm... looks like API stuff needs lots of stuff in Contextfied...

@patrickxb instead of API stuff (and everything) taking a "global" context... we should create an interfaces:

```go
type APIContext interface {
  Log  logger.Logger
  Env Env
}
```

And use that instead of Contextified (though can still pass in G()). That way it's much easier to test and use from external packages and libraries and doesn't create a massive dependency chain.

Basically anything that touches Contextified is poisoning the well.. and prevents any 3rd party package or library from using it unless it decides to create a context (which touches a lot)!

Anyway, probably too late to address now but maybe as a rule going forward, no more usage of Contextified for new stuff?

**Comment by MarcoPolo** [comment]
> Anyway, probably too late to address now but maybe as a rule going forward, no more usage of Contextified for new stuff?

How would I do that here?

**Pull request closed without merging**